crates/holochain_types/src/app.rs

```diff
@@ -231,9 +233,48 @@ impl Default for RoleSettings {
     }
 }
 
+impl From<RoleSettingsYaml> for RoleSettings {
+    fn from(role_settings: RoleSettingsYaml) -> Self {
+        match role_settings {
+            RoleSettingsYaml::Provisioned {
+                membrane_proof,
+                modifiers,
+            } => Self::Provisioned {
+                membrane_proof,
+                modifiers,
+            },
+            RoleSettingsYaml::UseExisting { cell_id } => Self::UseExisting { cell_id },
+        }
+    }
+}
+
+/// A version of RoleSettings that serializes to YAML without the content attribute
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum RoleSettingsYaml {
```

Comment:
new `RoleSettingsYaml` type here to allow for distinct serialization between reading from yaml vs. receiving it via conductor API.